include/deal.II/lac/lapack_full_matrix.h

```diff
@@ -57,7 +57,13 @@ class LAPACKFullMatrix : public TransposeTable<number>
   /**
    * Declare type for container size.
    */
-  typedef unsigned int size_type;
+  typedef types::blas_int size_type;
```

Comment:
I don't like this change. We've tried to be consistent in mapping the semantics of things into the type system. Indices and sizes can only be non-negative integers, so that's what we should try to keep with.

If not, then at least we should carefully explain why we deviate from our usual practice. (And we may have to add checks for `index >= 0` to all of the places where we pass indices as arguments.)

Reply:
As a compromise, we could use `types::blas_int` in conjnction the type trait [`std::make_unsigned`](http://en.cppreference.com/w/cpp/types/make_unsigned).

```c++
typedef typename std::make_unsigned<types::blas_int>::type size_type;
```

Reply:
Yes, sure. That seems reasonable.